### DISCUSSION TITLE
could it be taken further, to run in FireCracker VM, launched on demand

### DISCUSSION THREAD

**Discussion by urbien** [chadgeary/cloudblock#3] [Ideas]
Awesome devops with Cloudblock!
I described the goal here:
https://github.com/pirate/wireguard-docs/issues/64

**Comment by chadgeary** [comment]
What is "checks client is legit"....?

**Comment by urbien** [comment]
thanks for your reply. It could be against the list of your own device IPs, or the other way around - blacklisted IPs, checking for revoked keys in case of compromised devices, checking for bandwidth quota exceeded as most VPSes limit bandwidth, DDOS, or any other rules you would want to have
